src/db/add_package.rs

```diff
@@ -360,64 +379,51 @@ fn update_owners_in_database(
     owners: &[CrateOwner],
     crate_id: i32,
 ) -> Result<()> {
-    let rows = conn.query(
-        "
-            SELECT login
-            FROM owners
-            INNER JOIN owner_rels
-                ON owner_rels.oid = owners.id
-            WHERE owner_rels.cid = $1
-        ",
-        &[&crate_id],
+    // Update any existing owner data since it is mutable and could have changed since last
+    // time we pulled it
+    let owner_upsert = conn.prepare(
+        "INSERT INTO owners (login, avatar, name, email)
+        VALUES ($1, $2, $3, $4)
+        ON CONFLICT (login) DO UPDATE
+            SET
+                avatar = EXCLUDED.avatar,
+                name = EXCLUDED.name,
+                email = EXCLUDED.email",
```

Comment:
Would it make sense to have this `RETURNING id`, and collect up all the ids to use in the next query instead of looking up by `login` again?

Reply:
valid point, I'll update the statements